client/src/components/ShareButton/ShareButton.tsx

```diff
@@ -29,16 +35,32 @@ const CommonSocialMediaShareButton = ({ complete_url, media_icon }) => (
     {media_icon}
   </a>
 );
-export class ShareButton extends React.Component {
-  constructor(props) {
-    super();
 
+interface ShareButtonProps {
+  button_class_name: string;
+  button_description: string;
```

Comment:
I think any prop with a default value should be marked optional, double check that.

Reply:
I noticed there is a few more components with props that have a default value. 
Should I open another PR for this issue? Sometimes I get this error in the file when I change props from required to optional: `Argument of type 'placeholdertype | undefined' is not assignable to parameter of type 'placeholdertype'.`, is it fine with I fix the error by doing something like `String(variable)` since we can't use `!` to assert non-null? Or should I use if statement instead to make sure the variable is not null?

Reply:
Sounds like the React-ways of doing default prop values are somewhat incompatible with TS. Maybe move the default values in to the prop types or something so TS knows? Give it a google, must be a common problem, likely clear advice on how we should be doing this. If you find something, post it in slack (and yeah, either open an issue or PR to clean these cases up).

Either way, something like `String(variable)` is a bad hack, even if the default means it wouldn't happen\*, that'd just be converting `undefined` to `"undefined"`. Remember that TS doesn't exist at run-time, it's just a guide to help us write code that should have safer run-time behaviour. If you're writing something with potentially unwanted run-time behaviour just to make the type system happy, then it's a good sign that you're headed in the wrong direction.

\* wouldn't happen right now, but what if some code changes in the future or the default is dropped?

Reply:
https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/default_props/